rm/rm-node/src/main/java/org/ow2/proactive/resourcemanager/utils/RRDSigarDataStore.java

```diff
@@ -213,4 +199,86 @@ void sample(RrdDb dataBase, long timeInMs) throws IOException {
+            if (attrValue instanceof CompositeDataSupport && compositeTypes.get(fullName) != null) {
+                Object val = ((CompositeDataSupport) attrValue).get(compositeTypes.get(fullName));
+                if (val != null) {
+                    attrValue = val;
+                }
+            }
+            sample.setValue(dataSource, Double.parseDouble(attrValue.toString()));
+            if (logger.isTraceEnabled()) {
+                logger.trace(timeInMs / 1000 + " sampling: " + dataSource + " / " + fullName + " " +
+                             Double.parseDouble(attrValue.toString()));
+            }
+        } catch (NumberFormatException ex) {
+            // do not save non-numeric values
+            String exceptionAsString = StackTraceUtil.getStackTrace(ex).replaceAll("For input string: \".*\"", "");
+            exceptionAsString = fullName + " / " + exceptionAsString;
+
+            logErrorOnce(exceptionAsString,
+                         "Non numeric value for " + dataSource + " / " + fullName + ": " + ex.getMessage());
+        } catch (Exception e) {
+            String exceptionAsString = StackTraceUtil.getStackTrace(e);
+            logErrorOnce(exceptionAsString,
+                         "Error while reading attribute " + attrName + " for object " + objectName + ": ",
+                         e);
+        }
+    }
+
+    /**
+     * Logs a sigar error only once to avoid flooding
+     * @param exceptionAsString key used to memorize exceptions already handled
+     * @param message message to display
+     */
+    private void logErrorOnce(String exceptionAsString, String message) {
+        if (!exceptionsThrown.contains(exceptionAsString)) {
+            exceptionsThrown.add(exceptionAsString);
+            logger.warn(message);
+        } else if (logger.isTraceEnabled()) {
+            logger.trace(message);
+        }
+    }
+
+    /**
+     * Logs a sigar error only once to avoid flooding
+     * @param exceptionAsString key used to memorize exceptions already handled
+     * @param message message to display
+     * @param e real exception
+     */
+    private void logErrorOnce(String exceptionAsString, String message, Exception e) {
+        if (!exceptionsThrown.contains(exceptionAsString)) {
+            exceptionsThrown.add(exceptionAsString);
+            logger.warn(message, e);
+        } else if (logger.isTraceEnabled()) {
+            logger.trace(message, e);
+        }
+    }
+
+    private class StdErrRedirectionHelper {
+        private ByteArrayOutputStream storedErrors;
+
+        private PrintStream redirectedStream;
+
+        private PrintStream console;
+
+        public PrintStream redirectStdErr() throws UnsupportedEncodingException {
+            storedErrors = new ByteArrayOutputStream();
+            redirectedStream = new PrintStream(storedErrors, true, StandardCharsets.UTF_8.name());
+            console = System.err;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Replace this use of System.out or System.err by a logger. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS106)

Reply:
no I won't do that, dude ...